tests/robot-tests/scripts/permalink_snapshots/check_permalinks.py

```diff
@@ -0,0 +1,268 @@
+import csv
+import os
+import time
+
+import requests
+from bs4 import BeautifulSoup
+from requests import Response
+
+"""
+Script for checking the health of permalinks & whether they can be rendered.
+Some permalinks on production don't load i.e. https://explore-education-statistics.service.gov.uk/data-tables/permalink/aed871a3-61e5-4f2a-8a2b-cceeca680980
+This will cause issues when migrating to permalink snapshots (see https://dfedigital.atlassian.net/browse/EES-3784)
+
+Instructions:
+
+To generate the csv file of permalinks run the following query against the content DB:
+
+```
+SELECT *
+from [dbo].[Permalinks]
+```
+
+place these permalinks in a csv file in the same directory as '~/permalink-snapshots'
+with the filename 'prod-permalinks.csv'
+
+create a csv file with the name 'invalid-permalinks.csv' in the same directory as '~/permalink-snapshots'
+with the following headers:
+
+```
+permalink_id,subject_name,publication_name,status_code,has_table,created_date,out_of_date,error_message,warning_message,response_time
+```
+
+"""
+
+
+class PermalinkType:
+    id = str
+    publicationTitle = str
+    dataSetTitle = str
+    releaseId = str
+    subjectId = str
+    created = str
+
+
+requests.sessions.HTTPAdapter(pool_connections=100, pool_maxsize=100, max_retries=3)
+
+
+class PermalinkChecker:
```

Comment:
Nice work :smiley:

Reply:
ta, took a few tries to find the balance between the speed of the script vs the impact that had on the public frontend. Hopefully should be useful in getting a lay of the land for bad permalinks out there.